engine/sound/src/devices/device_wasapi.cpp

```diff
@@ -140,34 +140,53 @@ namespace dmDeviceWasapi
         }
 
         // While we can suggest the format, the SHARED mode will return the current format
-        WAVEFORMATEX format;
-        format.cbSize           = sizeof(WAVEFORMATEX);
-        format.wFormatTag       = WAVE_FORMAT_PCM;
-        format.nChannels        = 2;
-        format.nSamplesPerSec   = 44100; // Currently our supported audio format, so let's hint for that
-        format.wBitsPerSample   = 16;
-        format.nBlockAlign      = format.nChannels * (format.wBitsPerSample >> 3);
-        format.nAvgBytesPerSec  = format.nSamplesPerSec * format.nBlockAlign;
-
-        hr = device->m_AudioClient->IsFormatSupported(AUDCLNT_SHAREMODE_SHARED, &format, &device->m_MixFormat);
+        WAVEFORMATEX* pformat;
+        hr = device->m_AudioClient->GetMixFormat(&pformat);
         if (FAILED(hr))
         {
-            dmLogError("Format not supported");
+            dmLogError("Failed to get mix format from device:");
             CheckAndPrintError(hr);
             DeleteDevice(device);
             return dmSound::RESULT_INIT_ERROR;
         }
 
-        if (WAVE_FORMAT_EXTENSIBLE == device->m_MixFormat->wFormatTag)
+        dmLogInfo("Mix format:");
+        dmLogInfo("  wFormatTag:       %x  IEEE_FLOAT/PCM/EXTENSIBLE: %x / %x / %x", pformat->wFormatTag, WAVE_FORMAT_IEEE_FLOAT, WAVE_FORMAT_PCM, WAVE_FORMAT_EXTENSIBLE);
+        dmLogInfo("  nChannels:        %d", pformat->nChannels);
+        dmLogInfo("  nSamplesPerSec:   %d", pformat->nSamplesPerSec);
+        dmLogInfo("  nAvgBytesPerSec:  %d", pformat->nAvgBytesPerSec);
+        dmLogInfo("  nBlockAlign:      %d", pformat->nBlockAlign);
+        dmLogInfo("  wBitsPerSample:   %d", pformat->wBitsPerSample);
+
+        WAVEFORMATEX* closest = 0;
+        hr = device->m_AudioClient->IsFormatSupported(AUDCLNT_SHAREMODE_SHARED, pformat, &closest);
+        if (FAILED(hr))
+        {
+            dmLogError("Format not supported:");
+            CheckAndPrintError(hr);
+            if (closest)

```

Comment:
If for some reason, its suggested format wasn't available, it will return a "closest" suggestion, so let's use that instead.